client/sysinfo/src/sysinfo.rs

```diff
@@ -32,6 +33,64 @@ use std::{
 	time::{Duration, Instant},
 };
 
+/// Throughput as measured in bytes per second.
+#[derive(Deserialize, Serialize, Debug, Clone, Copy, PartialEq)]
+pub struct Throughput(f64);
+
+const KIBIBYTE: f64 = 1024.0;
+
+impl Throughput {
+	/// `f64` kibibyte/s to byte/s.
+	pub fn from_kibs(mibs: f64) -> Throughput {
+		Throughput(mibs * KIBIBYTE)
+	}
```

Comment:
Updated the `Throughput`, does this look reasonable?

Reply:
Yeah, it's better.

Few more things from the top of my head that can/should be improved:
- Telemetry's still broken.
- You should modify the `benchmark` function so that it returns a value with full precision. (And it can probably just return `Throughput`, I guess?)
- I'd rename the `to_kibs`/etc. functions to `as_kibs`/etc.. (That's how e.g. `std::time::Duration` names its functions; let's be consistent)
- The `to_kibs`/etc. should not return a `Throughput` anymore; `Throughput` internally should *always* be the same unit (that is, in bytes)